back-end/controllers/AuthController.js

```diff
@@ -31,6 +32,9 @@ exports.createUser = async (req, res) => {
 
     register(req.body);
     await createRoleForUser(req.body.role, req.body);
+    const { _id } = await findRoleId(req.body.email);
```

Comment:
You don't need to do find again, you can return these objects from register and createRoleForUser methods